docs/explanations/model_diagnostics/index.rst

```diff
@@ -0,0 +1,63 @@
+Model Diagnostics Workflow
+==========================
+
+.. contents::
+    :depth: 3
+    :local:
+
+Introduction
+------------
+
+Writing well-posed equation-oriented models is a significant challenge, and even the most experienced developers often have to spend a lot of time diagnosing and resolving issues before a model is able to solve reliably. This documentation is intended to assist users with this process by outlining a general workflow for model development and debugging in order to more easily identify and resolve modeling issues.
```

Comment:
Currently the link to the DiagnosticsToolbox API reference is all the way down at the bottom of this page. Maybe it would be helpful to add the sentence
```rst
A detailed description of the ``DiagnosticsToolbox`` API can be found :ref:`here<reference_guides/core/util/diagnostics/diagnostics_toolbox:Diagnostics Toolbox>`
```
to this paragraph.